engine/function/src/main/java/io/deephaven/function/Logic.java

```diff
@@ -49,8 +49,7 @@ static public Boolean and(boolean... values) {
      * @return logical and of all the values in the array. By convention, returns true if the array is empty.
      */
     static public Boolean and(ObjectVector<Boolean> values) {
-        for (int ii = 0; ii < values.size(); ++ii) {
-            Boolean b = values.get(ii);
+        for (final Boolean b : values) {
```

Comment:
One thing I did a poor job of explaining in the issue is that these iterators must be closed, although for all the current implementations it’s enough to use them until they are exhausted. This loop *is* using the iterator to exhaustion, so it's "OK". That said, if we wanted to be extra careful/pedantic, we could instead use a try-with-resources block. This applies to pretty much every usage, so just commenting once for discussion.

Reply:
We decided to use try-with-resources everywhere.